replica/database.hh

```diff
@@ -555,7 +552,7 @@ private:
     lw_shared_ptr<sstables::sstable_set>
     do_add_sstable(lw_shared_ptr<sstables::sstable_set> sstables, sstables::shared_sstable sstable,
         enable_backlog_tracker backlog_tracker);
-    void add_sstable(sstables::shared_sstable sstable);
+    void add_sstable(compaction_group& cg, sstables::shared_sstable sstable);
```

Comment:
I expected to see table::add_sstable(sstable) that figures out which compaction_group to talk to, and compaction_group::add_sstable(sstable) that does the work. How does the caller of add_sstable(compaction_group, sstable) know to select the correct compaction_group? Or to split the sstable if needed?

Reply:
table::add_sstable(compaction_group& cg, sstables::shared_sstable sstable) is a private function. I think you're referring to table::add_sstable_and_update_cache() which is in the public namespace. And yes, it should figure out which group a file belongs to. In this series, add_sstable_and_update_cache() is adding the sstable on behalf of the single group. With my next step, which is to add support to multiple groups, then a function will indeed be introduced to figure out the ownership of the sstable file.